.github/workflows/publish-runtime.yml

```diff
@@ -19,18 +19,46 @@ jobs:
     strategy:
       matrix:
         chain: ["moonbase", "moonriver", "moonbeam"]
+        srtool_image:
+          - purestake/srtool
+        srtool_image_tag:
+          - 1.62.0
     steps:
       - name: Get Timestamp
         run: echo "TMSP=$(date '+%Y%m%d_%H%M%S')" >> $GITHUB_ENV
       - uses: actions/checkout@v2
         with:
           ref: ${{ github.event.inputs.to }}
-      - name: Srtool build
-        id: srtool_build
-        uses: chevdor/srtool-actions@v0.5.0
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        if: env.GITHUB_REPOSITORY == "purestake/moonbeam"
         with:
-          chain: ${{ matrix.chain }}
-          tag: 1.62.0
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build & Push purestake/srtool image
+        if: env.GITHUB_REPOSITORY == "purestake/moonbeam"
+        run: |
+          docker pull "${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}" && image_exists=true || image_exists=false
+
+          if [[ $image_exists = "false" ]]; then
+            echo building "${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}"
+            docker build --pull --no-cache . \
+              -f docker/moonbeam-srtool.Dockerfile \
+              --build-arg SRTOOL_IMAGE_TAG=${{ matrix.srtool_image_tag }} \
+              -t ${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}
+
+            echo pushing "${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}"
+            docker push "${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}"
+          else
+            echo skiping build "${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}", image already exists
+          fi
+      - name: Build runtime using "${{ matrix.srtool_image }}:${{ matrix.srtool_image_tag }}"
+        id: srtool_build
+        run: |-
+          export GH_WORKFLOW_MATRIX_CHAIN=${{ matrix.chain }}
```

Comment:
Please use `env:` instead, it's more readable and more consistent with our GH workflows